core/src/stores/postgres.rs

```diff
@@ -1387,23 +1395,25 @@ impl Store for PostgresStore {
             .await?;
 
         let stmt = tx
+            // TODO @fontanierh: remove tags_json once we have tags_array everywhere
             .prepare(
                 "INSERT INTO data_sources_documents \
-                   (id, data_source, created, document_id, timestamp, tags_json, parents, \
+                   (id, data_source, created, document_id, timestamp, tags_json, tags_array, parents, \
                     source_url, hash, text_size, chunk_count, status) \
-                   VALUES (DEFAULT, $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11) RETURNING id",
+                   VALUES (DEFAULT, $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12) RETURNING id",
             )
             .await?;
 
-        let document_tags_data = serde_json::to_string(&document_tags)?;
+        let document_tags_json_string = serde_json::to_string(&document_tags)?;
         tx.query_one(
             &stmt,
             &[
                 &data_source_row_id,
                 &(document_created as i64),
                 &document_id,
                 &(document_timestamp as i64),
-                &document_tags_data,
+                &document_tags_json_string,
+                &document_tags,
```

Comment:
I would do that only after you're fully migrated in the same follow-up PR as the one that removes the updating of tags_json and the colum itself?

That way you can create the column, deploy and them migrate and have guarantee you have everything without risking returning [] during the migration?

Reply:
(essentially, I fill both fields for new rows, but keep using `tags_json` for now)